pkg/services/app/add.go

```diff
@@ -0,0 +1,503 @@
+package app
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+	"github.com/weaveworks/weave-gitops/pkg/git"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+)
+
+type DeploymentType string
+type SourceType string
+type ConfigType string
+
+const (
+	DeployTypeKustomize DeploymentType = "kustomize"
+	DeployTypeHelm      DeploymentType = "helm"
+
+	SourceTypeGit  SourceType = "git"
+	SourceTypeHelm SourceType = "helm"
+
+	ConfigTypeUserRepo ConfigType = ""
+	ConfigTypeNone     ConfigType = "NONE"
+)
+
+type AddParams struct {
+	Dir                  string
+	Name                 string
+	Url                  string
+	Path                 string
+	Branch               string
+	PrivateKey           string
+	PrivateKeyPass       string
+	DeploymentType       string
+	Chart                string
+	SourceType           string
+	CommitManifests      bool
+	AutomationRepo       string
+	AutomationRepoPath   string
+	AutomationRepoBranch string
+	Namespace            string
+	DryRun               bool
+}
+
+func (a *App) Add(params AddParams) error {
+	fmt.Print("Updating parameters from environment... ")
+	params, err := a.updateParametersIfNecessary(params)
+	if err != nil {
+		return errors.Wrap(err, "could not update parameters")
+	}
+
+	fmt.Print("done\n\n")
+	fmt.Print("Checking cluster status... ")
+
+	clusterStatus := a.kube.GetClusterStatus()
+	fmt.Printf("%s\n\n", clusterStatus)
+
+	switch clusterStatus {
+	case kube.Unmodified:
+		return errors.New("WeGO not installed... exiting")
```

Comment:
Minor log message improvement: not necessarily exiting in this case.

Reply:
Actually is, I was asked to add that at some point, and just exit there if we can't talk to the cluster.